xdsl/ir/core.py

```diff
@@ -888,14 +889,23 @@ def drop_all_references(self) -> None:
         for region in self.regions:
             region.drop_all_references()
 
-    def walk(self) -> Iterator[Operation]:
+    def walk(
+        self, *, reverse: bool = False, region_first: bool = False
+    ) -> Iterator[Operation]:
         """
-        Iterate all operations contained in the operation (including this one)
+        Iterate all operations contained in the operation (including this one).
+        If region_first is set, then the operation regions are iterated before the
+        operation. If reverse is set, then the region, block, and operation lists are
+        iterated in reverse order.
         """
-        yield self
-        for region in self.regions:
-            yield from region.walk()
+        if not region_first:
+            yield self
+        for region in self.regions[::-1] if reverse else self.regions:
```

Comment:
```suggestion
        for region in reversed(self.regions) if reverse else self.regions:
```
Would this work? `[::-1]` creates a new list

Reply:
Oh yes, thanks!